app/org/sagebionetworks/bridge/services/IntentService.java

```diff
@@ -133,6 +140,27 @@ public void registerIntentToParticipate(Study study, StudyParticipant participan
         }
     }
     
+    public boolean registerIntentToParticipate(Study study, Account account) {
+        Phone phone = account.getPhone();
+        // Somehow, this is being called but the user has no phone number.
+        if (phone == null) {
+            return false;
+        }
+        List<Subpopulation> subpops = subpopService.getSubpopulations(study.getStudyIdentifier());
+        for (Subpopulation subpop : subpops) {
+            CacheKey cacheKey = CacheKey.itp(subpop.getGuid(), study.getStudyIdentifier(), phone);
+            IntentToParticipate intent = cacheProvider.getObject(cacheKey, IntentToParticipate.class);
+            if (intent != null) {
+                StudyParticipant participant = participantService.getParticipant(study, account.getId(), true);
+                consentService.consentToResearch(study, subpop.getGuid(), participant, 
+                        intent.getConsentSignature(), intent.getScope(), true);
+                cacheProvider.removeObject(cacheKey);
+                return true;
```

Comment:
Could an ITP represent multiple consents? If so, instead of returning true and short-circuiting, would it be better to save the result somewhere and return it later?

Reply:
Theoretically it could. Changing this, also keeping track as to whether or not we've loaded the study participant.